src/core/positioning/abstractgnssreceiver.h

```diff
@@ -55,20 +54,24 @@ class AbstractGnssReceiver : public QObject
 
     GnssPositionInformation lastGnssPositionInformation() const { return mLastGnssPositionInformation; }
 
-    QAbstractSocket::SocketState socketState() const { return mSocketState; }
-    QString socketStateString() const { return mSocketStateString; }
     QString lastError() const { return mLastError; }
 
     Q_INVOKABLE virtual AbstractGnssReceiver::Capabilities capabilities() const { return NoCapabilities; }
 
-    virtual QList<QPair<QString, QVariant>> details() { return {}; }
+    virtual QList<QPair<QString, QVariant>> details() const { return {}; }
+    virtual QAbstractSocket::SocketState socketState() const { return mSocketState; }
+
+
+  public slots:
+    virtual QString socketStateString();
+    void setSocketState( const QAbstractSocket::SocketState &state );
```

Comment:
Make socketStateString public, and make setSocketState protected (we dont want to expose the latter basically)

Reply:
I would say that `setSocketState` should not even exitst. Or at most as implementation on subclasses if they need that

Reply:
and mSocketState as well, otherwise there is only the risk that it gets out of sync with `subclass::socketState`